# Parameters for adaptive_grasping package
# Remember that there can be other important constants that can only be changed
# from the src code.

adaptive_grasping:
  # The topic where the finger touches data is published
  touch_topic_name: "/touching_finger_topic"
  # The following map contains the correspondance between ids and link names
  link_names_map:
    "1" : "right_hand_thumb_distal_link"
    "2" : "right_hand_index_distal_link"
    "3" : "right_hand_middle_distal_link"
    "4" : "right_hand_ring_distal_link"
    "5" : "right_hand_little_distal_link"
  # The following map contains other necessary params for adaptive_grasping
  params_map:
    world_name: "world"
    palm_name: "right_hand_ee_link"
    ee_name: "right_hand_ee_link"
  # The array containing the number of links of each finger
  joints_num: [5, 7, 7, 7, 7]
  # The following list of arrays is the contact selection matrix H (CHANGE ALSO p_vector ACCORDINGLY)
  h_matrix:     # fully constrained
    - [1, 0, 0, 0, 0, 0]
    - [0, 1, 0, 0, 0, 0]
    - [0, 0, 1, 0, 0, 0]
    - [0, 0, 0, 1, 0, 0]
    - [0, 0, 0, 0, 1, 0]
    - [0, 0, 0, 0, 0, 1]
  h_matrix_2:   # position constrained
    - [1, 0, 0, 0, 0, 0]
    - [0, 1, 0, 0, 0, 0]
    - [0, 0, 1, 0, 0, 0]
  # The following list of arrays is the minimization weight matrix A tilde
  a_tilde_matrix:
    - [100, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    - [0, 10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    - [0, 0, 10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    - [0, 0, 0, 10, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    - [0, 0, 0, 0, 10, 0, 0, 0, 0, 0, 0, 0, 0]
    - [0, 0, 0, 0, 0, 10, 0, 0, 0, 0, 0, 0, 0]
    - [0, 0, 0, 0, 0, 0, 10, 0, 0, 0, 0, 0, 0]
    - [0, 0, 0, 0, 0, 0, 0, 10000, 0, 0, 0, 0, 0]
    - [0, 0, 0, 0, 0, 0, 0, 0, 10000, 0, 0, 0, 0]
    - [0, 0, 0, 0, 0, 0, 0, 0, 0, 10000, 0, 0, 0]
    - [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10000, 0, 0]
    - [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10000, 0]
    - [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10000]
  # The following array is the desired x motion
  x_d:
    # - [0.1, 0.01, 0, -0.01, 0, -0.05, 0, 0, 0, 0, 0, 0, 0]      # for other fingers mainly?
    - [0.1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]                   # for simple grasp?
    # - [0.1, 0, 0, -0.01, 0.05, 0, 0, 0, 0, 0, 0, 0, 0]          # for thumb mainly?
    # - [0.5, 0.00, 0, -0.05, 0, -0.09, 0, 0, 0, 0, 0, 0, 0]      # random stuff 1
    # - [0.5, 0.00, 0, -0.05, 0, 0.00, 0, 0, 0, 0, 0, 0, 0]       # random stuff 2
  x_d_2:
    - [0.0, 0.01, 0, -0.01, 0, -0.05, 0, 0, 0, 0, 0, 0, 0]      # for two finger contacts
  # The rate of the adaptive loop
  spin_rate: 1000.0
  # The topic where the object pose is published
  object_topic_name: "/object_pose"
  # The scaling factor for the reference motion
  scaling: 1.0
  # The permutation vector for relaxed minimization (the elements of the vector stand for the reordering of the rows of Q) (CHANGE ALSO h_matrix ACCORDINGLY)
  p_vector:
    - [6, 3, 4, 5, 1, 2]              # fully constrained
  p_vector_2:
    - [2, 3, 1]                       # position constrained
  # The vector containing the indexes of the touch constraints
  touch_indexes:
    - [14, 15, 16, 17, 18, 19]
  # The threshold on the synergy joint value for stopping the grasping motion
  syn_thresh: 0.8
  # The bool to decide if null twist should be sent when relaxation occurs
  relax_to_zero: false
  # The bool to decide if to change contact selection when touch occurs
  touch_change: true


